test/test_gradient.jl

```diff
@@ -21,7 +21,7 @@ dϕ = dϕ/norm(dϕ) * norm(ϕ)/2.75e11
 
 @testset "Taylor-series gradient test of jutulModeling with wells" begin
     grad_test(x0->misfit(x0, ϕ, q, states), x0, dx, g[x0])
-    grad_test(ϕ->misfit(x0, ϕ, q, states), ϕ, dϕ, g[ϕ])
+    grad_test(ϕ->misfit(x0, ϕ, q, states), ϕ, dϕ, g[ϕ], h0=2e1, maxiter=12)
```

Comment:
why this change? Might be why it fails as larger h0 tend to not be asymptotic enough

Reply:
This test is passing. The vertical wells test later in this file is failing.

I had to increase the step size to get this test to pass. It seemed that the step size was too small for this and the other porosity gradients. But increasing the step size didn't help for the failing vertical wells test.